File: YoonJunSik/week_5/조이스틱.java
class Solution {
    public int solution(String name) {
        int len = name.length();
        int answer = 0;

        // 1. 상하 이동 최소 횟수
        for(int i = 0; i < len; i++) {
            char c = name.charAt(i);
            int up = c - 'A';
            int down = 'Z' - c + 1;
            answer += Math.min(up, down);
        }

        // 2. 좌우 이동 최소 횟수
        int move = len - 1; // 기본 직진 이동

        for(int i = 0; i < len; i++)
        {
            int next = i + 1;

            // A가 연속되는 구간을 찾기
            while(next < len && name.charAt(next) == 'A') {
                next++;
            }

            // 되돌아가기 방식 중 최소값으로 갱신
            move = Math.min(move, i * 2 + (len - next)); // 오른쪽 → 왼쪽 돌아오기
            move = Math.min(move, (len - next) * 2 + i); // 왼쪽 → 오른쪽 돌아오기
        }
        answer += move;
        return answer;
    }
}

File: ParkJuHyun/week_5/조이스틱.py
def solution(name):
    answer = 0
    
    for c in name:
        answer += min(ord(c)-ord('A'), 26 - (ord(c)-ord('A')))

    n = len(name)
    min_move = n-1
    
    for i in range(n):
        next_idx = i+1
        while next_idx < n and name[next_idx] == 'A':
            next_idx += 1
        
        distance = min(i,n - next_idx)
        min_move = min(min_move, i + n - next_idx + distance)
    
    answer += min_move
    return answer